src/Features/Core/Completion/Providers/AbstractSymbolCompletionProvider.cs

```diff
@@ -19,6 +19,14 @@ namespace Microsoft.CodeAnalysis.Completion.Providers
 {
     internal abstract partial class AbstractSymbolCompletionProvider : AbstractCompletionProvider
     {
+        // PERF: Many CompletionProviders dervice AbstractSymbolCompletionProvider and therefore
+        // compute identical contexts. This actually shows up on the 2-core typing test.
+        // Cache the most recent document/position/computed SyntaxContext to reduce repeat computation.
+        private static Document cachedDocument;
+        private static int cachedPosition;
+        private static AbstractSyntaxContext cachedContext;
+        private static readonly object cacheGate = new object();
```

Comment:
@rchande Yes, but not as it exists in the current sources. Look at Iteration 1 of the CodeFlow and the _keywordListCache field.